File: SDKs/googleapis.com/sheets/v4/python/src/sdk/models/shared/batchupdatevaluesresponse.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from . import updatevaluesresponse


@dataclass_json
@dataclass
class BatchUpdateValuesResponse:
    responses: Optional[List[updatevaluesresponse.UpdateValuesResponse]] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'responses' }})
    spreadsheet_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'spreadsheetId' }})
    total_updated_cells: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalUpdatedCells' }})
    total_updated_columns: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalUpdatedColumns' }})
    total_updated_rows: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalUpdatedRows' }})
    total_updated_sheets: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'totalUpdatedSheets' }})
    

File: SDKs/googleapis.com/sheets/v4/typescript/src/sdk/models/shared/batchupdatevaluesresponse.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { UpdateValuesResponse } from "./updatevaluesresponse";



// BatchUpdateValuesResponse
/** 
 * The response when updating a range of values in a spreadsheet.
**/
export class BatchUpdateValuesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=responses", elemType: UpdateValuesResponse })
  responses?: UpdateValuesResponse[];

  @SpeakeasyMetadata({ data: "json, name=spreadsheetId" })
  spreadsheetId?: string;

  @SpeakeasyMetadata({ data: "json, name=totalUpdatedCells" })
  totalUpdatedCells?: number;

  @SpeakeasyMetadata({ data: "json, name=totalUpdatedColumns" })
  totalUpdatedColumns?: number;

  @SpeakeasyMetadata({ data: "json, name=totalUpdatedRows" })
  totalUpdatedRows?: number;

  @SpeakeasyMetadata({ data: "json, name=totalUpdatedSheets" })
  totalUpdatedSheets?: number;
}
